lib/fluent/plugin/out_elasticsearch.rb

```diff
@@ -349,12 +357,93 @@ def get_parent_of(record, path)
     [parent_object, path[-1]]
   end
 
-  def send_bulk(data)
+  def send_bulk(data, count)
     retries = 0
     begin
       response = client.bulk body: data
       if response['errors']
-        log.error "Could not push log to Elasticsearch: #{response}"
+        errors = Hash.new(0)
+        errors_bad_resp = 0
+        errors_unrecognized = 0
+        successes = 0
+        duplicates = 0
+        bad_arguments = 0
+        # Count up the individual error types returned for each item
+        # Note: this appears to be an undocumented feature of Elasticsearch
+        # https://www.elastic.co/guide/en/elasticsearch/reference/2.4/docs-bulk.html
+        # When you submit an "index" write_operation, with no "_id" field in the
+        # metadata header, Elasticsearch will turn this into a "create"
+        # operation in the response.
+        response['items'].each do |item|
+          if item.has_key?(@write_operation)
```

Comment:
case statment doest require a common evaluation like a switch statement.  It will evaluate each conditional.  Stupid example:

```
case
when 1==2
   puts 'a'
when [1].include?(2)
  puts "b"
else
 puts 'c'
end
``